terraform/mesh_aws/kms.tf

```diff
@@ -0,0 +1,96 @@
+resource "aws_kms_key" "mesh" {
+  description             = local.name
+  deletion_window_in_days = 14
+  is_enabled              = true
+  enable_key_rotation     = true
+  policy                  = data.aws_iam_policy_document.mesh.json
+}
+
+resource "aws_kms_alias" "mesh" {
+  name          = "alias/${local.name}"
+  target_key_id = aws_kms_key.mesh.key_id
+}
+
+data "aws_iam_policy_document" "mesh" {
+  statement {
+    sid       = "Enable IAM User Permissions"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["kms:*"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+  }
+
+  statement {
+    sid       = "Allow access for Key Administrators"
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "kms:Create*",
+      "kms:Describe*",
+      "kms:Enable*",
+      "kms:List*",
+      "kms:Put*",
+      "kms:Update*",
+      "kms:Revoke*",
+      "kms:Disable*",
+      "kms:Get*",
+      "kms:Delete*",
+      "kms:TagResource",
+      "kms:UntagResource",
+      "kms:ScheduleKeyDeletion",
+      "kms:CancelKeyDeletion",
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/NHSDAdminRole"]
```

Comment:
this is making an assumption that "NHSDAdminRole" will exist in target account. If 3rd parties use this terraform module, that may not be true. Could be passed through as module config?